public/app/plugins/datasource/elasticsearch/configuration/DataLinks.test.tsx

```diff
@@ -1,84 +1,64 @@
-import { mount } from 'enzyme';
+import { render, screen, fireEvent } from '@testing-library/react';
 import React from 'react';
-import { act } from 'react-dom/test-utils';
 
-import { Button } from '@grafana/ui';
+import { DataLinkConfig } from '../types';
 
-import { DataLink } from './DataLink';
-import { DataLinks } from './DataLinks';
+import { DataLinks, Props } from './DataLinks';
+
+const setup = (propOverrides?: Props) => {
+  const props: Props = {
+    value: [],
+    onChange: jest.fn(),
+  };
+
+  Object.assign(props, propOverrides);
+
+  return render(<DataLinks {...props} />);
+};
 
 describe('DataLinks', () => {
-  let originalGetSelection: typeof window.getSelection;
-  beforeAll(() => {
-    originalGetSelection = window.getSelection;
-    window.getSelection = () => null;
-  });
+  it('should render correctly with no fields', () => {
+    setup();
 
-  afterAll(() => {
-    window.getSelection = originalGetSelection;
+    expect(screen.getByRole('heading', { name: 'Data links' }));
+    expect(screen.getByRole('button', { name: 'Add' })).toBeInTheDocument();
   });
 
-  it('renders correctly when no fields', async () => {
-    let wrapper: any;
-    await act(
-      // @ts-ignore we shouldn't use Promises in act => the "void | undefined" is here to forbid any sneaky "Promise" returns.
-      async () => {
-        wrapper = await mount(<DataLinks onChange={() => {}} />);
-      }
-    );
-    expect(wrapper.find(Button).length).toBe(1);
-    expect(wrapper.find(Button).contains('Add')).toBeTruthy();
-    expect(wrapper.find(DataLink).length).toBe(0);
-  });
+  it('should render correctly when passed fields', () => {
+    setup({ value: testValue, onChange: () => {} });
 
-  it('renders correctly when there are fields', async () => {
-    let wrapper: any;
-    await act(
-      // @ts-ignore we shouldn't use Promises in act => the "void | undefined" is here to forbid any sneaky "Promise" returns.
-      async () => {
-        wrapper = await mount(<DataLinks value={testValue} onChange={() => {}} />);
-      }
-    );
-
-    expect(wrapper.find(Button).filterWhere((button: any) => button.contains('Add')).length).toBe(1);
-    expect(wrapper.find(DataLink).length).toBe(2);
+    expect(screen.getByText('localhost1')).toBeInTheDocument();
+    expect(screen.getByText('localhost2')).toBeInTheDocument();
+    expect(screen.getAllByRole('button', { name: 'Remove field' })).toHaveLength(2);
+    expect(screen.getAllByRole('checkbox', { name: 'Internal link' })).toHaveLength(2);
+
+    jest.spyOn(console, 'error').mockImplementation();
```

Comment:
Not sure why but for some reason when using 'getByAll' it's throwing an error and this is the only work-around I could find.

Reply:
Apparently there are some state changes/renders that are not completed when component is queried. In this case instead of `getAllBy`, you can use `findAllBy`, which is async and resolves when an element is found - `expect(await screen.findAllByRole('button', { name: 'Remove field' })).toHaveLength(2);`

Reply:
Ahh amazing,  Thanks!